CliFx/Schema/BindablePropertyDescriptor.cs

```diff
@@ -21,7 +21,17 @@ internal class BindablePropertyDescriptor : IPropertyDescriptor
 
         public IReadOnlyList<object?> GetValidValues()
         {
-            var underlyingType = Type.TryGetNullableUnderlyingType() ?? Type;
+            Type typeToCheck = Type;
+            foreach (var inf in typeToCheck.GetInterfaces())
+            {
+                if (inf.IsGenericType && inf.GetGenericTypeDefinition() == typeof(IList<>))
+                {
+                    typeToCheck = inf.GenericTypeArguments[0];
+                    break;
+                }
+            }
+
+            var underlyingType = typeToCheck.TryGetNullableUnderlyingType() ?? typeToCheck;
```

Comment:
We have `TryGetEnumerableUnderlyingType(...)`, can probably use that here instead of this?

https://github.com/Tyrrrz/CliFx/blob/d7460244b7899402d95fd06d8baee241d1549ff6/CliFx/Utils/Extensions/TypeExtensions.cs#L16-L33

something like

```csharp
var underlyingType =
    Type.TryGetNullableUnderlyingType() ??
    Type.TryGetEnumerableUnderlyingType() ??
    Type;
```

Reply:
Your method won't work most of the time, I suspect, because the type definition equality check only checks *direct* inheritance type. Indirect inheritance types require the loop that I have in my solution. I don't think this is going to be a functional replacement.

Another related question: Would you like to support `IEnumerable<>` as opposed to `IList<>`? Semantically I'm not sure if it matters, I know that the container semantics are different between the two. I'm just not entirely familiar with how generic your "container of arguments" support is. Let me know and I'm happy to change to `IEnumerable<>` if you prefer. Ideally this should be implemented in terms of the collection interfaces you already support for normal multi-argument options.

Reply:
I went ahead and switched it to `IEnumerable<>` anyway; test still passes. Let me know if this is good enough.

Reply:
`TryGetEnumerableUnderlyingType` can be used

Reply:
I apologize, I just realized I missed the bottom part of Tyrrz's code snippet from earlier. For some reason Github didn't make it obvious that I could scroll in there, so I didn't see the LINQ expression. It does look like that would indeed work. I also see that it was changed to use that solution on `master`.

However, I am realizing that none of the solutions up to this point work completely with nullable types. For example, `List<CustomEnum?>`. There's also a lack of test cases for the nullable case. I have solved this in another branch and I will open another PR shortly to address this.

I also am now using the solution here, but slightly different, to make sure that we get the nullable underlying type in more cases. Thank you once again for the feedback.